### ISSUE TITLE
help!

### ISSUE THREAD

**Issue by ashishkumarti** [grammarly/gector#9]
sir please help me to solve this error. (also, do we need to preprocess the official .m2 files downloaded from bea2019 website
![errorgec](https://user-images.githubusercontent.com/37670459/85957686-1faaf600-b9ad-11ea-8169-5228e949e225.JPG)
 )

**Comment by komelianchuk** [comment]
Hi!
You need:
1. Convert data from `m2` format into plain text. (pairs of source/target sentences)
2. Preprocess sentences by using `preprocess.py` script.
3. Split you preprocessed data into 98/2 split.
4. Run `train.py` script.

**Comment by ashishkumarti** [comment]
Thank you very much sir!!!

**Comment by sevilcaliskan** [comment]
Hi, This is probably a basic question and I don't want to open a new issue but I am a bit confused about converting m2 files into txt. It seems like I should use utils.prepare_clc_fce_data.py do create parallel sentences from m2 files and then use utils.preprocess. However, prepare_clc_fce_data seems to be taking .xml files as parameters. So, should I convert m2 files or am I doing something wrong?

**Comment by komelianchuk** [comment]
Hi.
Yes, you should convert m2 files to txt. Basically you need to extract only raw sentences without annotations.
I have not found the existing tool for doing that, but you can just read the file and extract all lines which start from `S `

**Comment by sevilcaliskan** [comment]
Thank you for your answer, I am new to GCE and m2 files. I can do that and those lines would be source sentences, right? What would be the target sentences?

**Comment by komelianchuk** [comment]
Can you clarify what dataset you want to convert into parallel files?
There is no need in doing that for evaluation sets (you just need source text, m2 file, and file with predictions in order to get scores)
And If I'm not mistaken training datasets are already available in parallel format (except FCE, but we provided a script for parsing it).
But it better to check here:
https://www.cl.cam.ac.uk/research/nl/bea2019st/

**Comment by sevilcaliskan** [comment]
I want to use FCE corpus for fine-tuning and I tried to use the script you provided but it seems to take .xml files as parameters and I am not sure how to proceed. I can't find a parallel format for Lang-8 either, but I will check again later in case I missed it. I know how to proceed with the other datasets now, thank you for replying in a short notice.

**Comment by komelianchuk** [comment]
I see.
Sorry, but I can not help with that right now, but I think @atrasevych will answer you after get back from vacation.

**Comment by sevilcaliskan** [comment]
OK, thank you, have a nice day.

**Comment by atrasevych** [comment]
Hi @sevilcaliskan!
Correct! The script that we have provided in the GECTOR repository is working with the XML format (the FCE corpus in XML format can be downloaded from https://ilexir.co.uk/datasets/index.html). The script takes two arguments: 1) path to the`fce-released-dataset` and 2) the output folder, where the parallel data will be saved.

As @komelianchuk previously mentioned, you can also use the FCE corpus (BEA version) from https://www.cl.cam.ac.uk/research/nl/bea2019st/ (it's m2 format). You can use `corr_from_m2.py` script (provided by BEA) to get the corrected version of the text from m2. To extract the original text, you can just read the file and extract all lines which start from `S`.

**Comment by sevilcaliskan** [comment]
OK, I will check the links now, thank you both very much for the answers.

**Comment by enkaranfiles** [comment]
> Can you clarify what dataset you want to convert into parallel files?
> There is no need in doing that for evaluation sets (you just need source text, m2 file, and file with predictions in order to get scores)
> And If I'm not mistaken training datasets are already available in parallel format (except FCE, but we provided a script for parsing it).
> But it better to check here:
> https://www.cl.cam.ac.uk/research/nl/bea2019st/

ı cannot find the other set in paralel file format. can you provide a link?
